src/libecalc/presentation/json_result/mapper.py

```diff
@@ -79,49 +78,43 @@ def get_requested_compressor_pressures(
     """
 
     evaluated_temporal_energy_usage_models = {}
-    default_date = datetime(1900, 1, 1, 0, 0)
-
-    # Extract relevant temporal model:
-    if len(energy_usage_model.items()) == 1 and default_date in energy_usage_model:
-        model_subset = TemporalModel(energy_usage_model).models[0].model
-    else:
-        model_subset = TemporalModel({model_period.start: energy_usage_model[model_period.start]}).models[0].model
-
-    if isinstance(model_subset, CompressorSystemConsumerFunction):
-        # Loop timesteps in temporal model, to find correct operational settings used:
-        for timestep in model_timesteps:
-            for compressor in model_subset.compressors:
-                if compressor.name == name:
-                    operational_setting_used_id = get_operational_setting_used_id(
-                        timestep=timestep, operational_settings_used=operational_settings_used
-                    )
 
-                    operational_setting = model_subset.operational_settings[operational_setting_used_id]
+    for period, model in TemporalModel(energy_usage_model).items():
+        if isinstance(model, CompressorSystemConsumerFunction):
+            # Loop timesteps in temporal model, to find correct operational settings used:
+            for timestep in model_timesteps:
+                if timestep >= period.start and timestep <= period.end:
```

Comment:
why is this test needed? Can we get timesteps "outside" the temporal model here? What if that happens? Is that an error (and should be handled) or is it normal and "proceed as normal"?

Reply:
Yes, `model_timesteps` are all timesteps for the component. The test is done in order to ensure that we are within the temporal model. The point is to grab the correct operational setting for the actual timestep.

Reply:
do we have a method like `period.get_timesteps(include_start=True, include_end=True)` that we can use instead to iterate? to avoid iterating _all_ timesteps from global start to global end for all temporal models?

Reply:
A new `get_timesteps` method is now included in `period`.